db/model/roomType.js

```diff
@@ -69,13 +74,122 @@ module.exports = function roomType(seq, dataTypes) {
         assoc.type = RoomType.hasMany(models.Room, {
           foreignKey: 'type',
         });
+        assoc.bots = RoomType.belongsToMany(models.Bot, {
+          foreignKey: 'roomTypeId',
+          through: 'RoomTypeBots',
+        });
         assoc.writers = RoomType.belongsToMany(models.User, {
           as: 'writers',
           through: 'RoomTypeWriters',
           foreignKey: 'roomTypeId',
         });
       },
     },
+    hooks: {
+
+      /**
+       * Ensures that all bots in request actually exist
+       *
+       * @param {Instance} inst - The instance being created
+       * @returns {Promise} which resolves to the instance, or rejects if
+       *  a bot is not found or duplicate bots are requested
+       */
+      beforeCreate(inst /* , opts */) {
+        return u.validateBotsArray(inst, seq);
+      }, // hooks.beforeCreate
+
+      /**
+       * Ensures that all bots in request actually exist
+       *
+       * @param {Instance} inst - The instance being created
+       * @returns {Promise} which resolves to the instance, or rejects if
+       *  a bot is not found or duplicate bots are requested
+       */
+      beforeUpdate(inst /* , opts */) {
+        return u.validateBotsArray(inst, seq);
+      }, // hooks.beforeUpdate
+
+      /**
+       * Deletes previous relationships with old bots and creates
+       * new relationships with new bots
+       *
+       * @param {RoomType} inst - The newly-created instance
+       */
+      afterUpdate(inst /* , opts */) {
+        const bots = inst.dataValues.bots;
+
+        return new seq.Promise((resolve, reject) => {
+          if (!bots || !inst.changed('bots')) {
+            resolve(inst);
+          }
+
+          inst._previousDataValues.bots.forEach((botName) => {
+            seq.models.Bot.findOne({ where: { name: botName } })
```

Comment:
same question about case-insensitivity everywhere you're using "name equals botName" in your finder